app/src/main/java/ch/epfl/favo/notifications/FirebaseMessagingService.java

```diff
@@ -0,0 +1,84 @@
+package ch.epfl.favo.notifications;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import com.google.firebase.messaging.RemoteMessage;
+
+import java.util.Objects;
+import java.util.Random;
+
+import ch.epfl.favo.MainActivity;
+import ch.epfl.favo.R;
+
+public class FirebaseMessagingService
+    extends com.google.firebase.messaging.FirebaseMessagingService {
+
+  //private static final String TAG = "MyFirebaseMsgService";
+  public static String CHANNEL_NAME = "Default channel name";
+
+  // method called when new message (notification or data message) is received
+  @Override
+  public void onMessageReceived(RemoteMessage remoteMessage) {
+
+    // Check if message contains a data payload.
+//    if (remoteMessage.getData().size() > 0) {
+//      Log.d(TAG, "Message data payload: " + remoteMessage.getData());
+//      // do something with message data, like chat (TODO for later sprints)
+//    }
+
+    // Check if message contains a notification payload.
+    if (remoteMessage.getNotification() != null) {
+      showNotification(this, remoteMessage.getNotification(), getString(R.string.default_notification_channel_id));
+    }
+  }
+
+  // onNewToken callback fires whenever a new token is generated
+  @Override
+  public void onNewToken(@NonNull String token) {
+    //Log.d(TAG, "Refreshed token: " + token);
+    // TODO send new refreshed token to db
+  }
+
+  // show notification received
+  public static void showNotification(Context context, RemoteMessage.Notification notification, String channelId) {
+    Intent intent = new Intent(context, MainActivity.class);
```

Comment:
Not sure if we should rather direct to the map and add the token on the map

Reply:
It actually redirects to the map, the map is on MainActivity. What do you mean by adding the token on the map?